File: backend/api/src/main/java/com/coinai/api/account/entity/Account.java
package com.coinai.api.account.entity;

import com.coinai.api.account.AccountType;
import com.coinai.api.user.entity.User;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "accounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Account {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 100)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AccountType type;

    @Column(nullable = false, length = 10)
    private String currency;

    @Column(nullable = false)
    private boolean archived;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

}

File: mobile/lib/features/account/domain/entities/account.dart
class Account {
  final String id;
  final String name;
  final String type;
  final String currency;
  final bool archived;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Account({
    required this.id,
    required this.name,
    required this.type,
    required this.currency,
    required this.archived,
    required this.createdAt,
    required this.updatedAt,
  });
}